src/core/qgsuserconfigmanager.h

```diff
@@ -0,0 +1,33 @@
+#ifndef QGSUSERCONFIGMANAGER_H
+#define QGSUSERCONFIGMANAGER_H
+
+#include <qgis.h>
+
+class CORE_EXPORT QgsUserConfig
+{
+  public:
+    QgsUserConfig( QString folder );
+    const QString folder() const;
+
+  private:
+    QString mConfigFolder;

```

Comment:
Is this expected to gain more information than the config folder? If not, could it be dumped and a simple string used instead?